nano/node/transport/tcp_server.cpp

```diff
@@ -82,15 +89,25 @@ void nano::transport::tcp_listener::stop ()
 			}
 		}
 		this_l->connections_per_address.clear ();
-	}));
+	});
 }
 
 std::size_t nano::transport::tcp_listener::connection_count ()
 {
 	nano::lock_guard<nano::mutex> lock{ mutex };
+	cleanup ();
```

Comment:
This seems like an odd place to do cleanup.

Reply:
Yes, it's less than ideal, but this is an intermediate step before doing larger refactoring.